src/core/runtime.cpp

```diff
@@ -625,16 +625,27 @@ SHWireState suspend(SHContext *context, double seconds) {
 
 void hash_update(const SHVar &var, void *state);
 
-std::unordered_set<const SHWire *> &gatheringWires() {
-#ifdef WIN32
-  // we have to leak.. or windows tls emulation will crash at process end
-  thread_local std::unordered_set<const SHWire *> *wires = new std::unordered_set<const SHWire *>();
-  return *wires;
-#else
-  thread_local std::unordered_set<const SHWire *> wires;
-  return wires;
-#endif
-}
+#define SH_HASHER(prefix)                                                               \
+  std::deque<std::unordered_set<const SHWire *>> &prefix##WiresStack() {                \
+    thread_local std::deque<std::unordered_set<const SHWire *>> s;                      \
+    return s;                                                                           \
+  }                                                                                     \
+  std::optional<std::unordered_set<const SHWire *> *> &prefix##Wires() {                \
```

Comment:
Might as well make this return a reference to the set directly (and assert internally that it's not null)
Especially since the usage below implies doing the same